api/namex/resources/requests.py

```diff
@@ -388,57 +388,28 @@ def get():
         example: query=NR3742302 or query=abcd
         """
         data = []
-        start = request.args.get('start', 0, type=int)
-        rows = request.args.get('rows', 10, type=int)
-        query = request.args.get('query', '')
-        if not query:
-            return make_response(jsonify(data), 200)
+        nr_num = request.args.get('query', '').strip().upper()
 
         try:
-            solr_query, nr_number, nr_name = SolrQueries.get_parsed_query_name_nr_search(query)
-            condition = ''
-            if nr_number:
-                condition = f"requests.nr_num ILIKE '%{nr_number}%'"
-            if nr_name:
-                nr_name = nr_name.replace("'", "''")
-                if condition:
-                    condition += ' OR '
-                name_condition = "requests.name_search ILIKE '%"
-                name_condition += "%' AND requests.name_search ILIKE '%".join(nr_name.split())
-                name_condition += "%'"
-
-                condition += f'({name_condition})'
-
-            results = RequestDAO.query.filter(
-                RequestDAO.stateCd.in_([State.DRAFT, State.INPROGRESS, State.REFUND_REQUESTED]),
-                text(f'({condition})')
+            # Find the NR only if it is not cancelled & not expired or if it has expired within a 60 day grace period. 
+            query = RequestDAO.query.filter(
+                RequestDAO.nrNum == nr_num,
+                RequestDAO.stateCd != State.CANCELLED,
+                or_(
+                    RequestDAO.stateCd != State.EXPIRED,
+                    text(f"(requests.state_cd = '{State.EXPIRED}' AND CAST(requests.expiration_date AS DATE) + "
+                            "interval '60 day' >= CAST(now() AS DATE))")
```

Comment:
This logic is now directly in the method. It used to call it here:
https://github.com/bcgov/namex/blob/main/api/namex/resources/requests.py#L458